src/test/integrasjonstester/kotlin/no/nav/familie/ks/sak/api/BehandlingControllerTest.kt

```diff
@@ -0,0 +1,162 @@
+        every { integrasjonClient.sjekkTilgangTilPersoner(any()) } returns Tilgang(true, "test")
+
+        Given {
+            header("Authorization", "Bearer $token")
+        } When {
+            get("$behandlingControllerUrl/${behandling.id}")
+        } Then {
+            statusCode(HttpStatus.OK.value())
+            body("data.behandlingId", Is(behandling.id.toInt()))
+            body("data.årsak", Is("SØKNAD"))
+        }
+    }
+
+    @Test
+    fun `endreBehandlendeEnhet - skal kaste FunksjonellFeil hvis begrunnelse er tom`() {
+        val token = lokalTestToken(behandlerRolle = BehandlerRolle.BESLUTTER)
+        every { integrasjonClient.sjekkTilgangTilPersoner(any()) } returns Tilgang(true, "test")
+
+        Given {
+            header("Authorization", "Bearer $token")
+            contentType(ContentType.JSON)
+            body(objectMapper.writeValueAsString(EndreBehandlendeEnhetDto("enhet", "")))
+        } When {
+            put("$behandlingControllerUrl/${behandling.id}/enhet")
+        } Then {
+            body("status", Is("FUNKSJONELL_FEIL"))
+            body("melding", Is("Begrunnelse kan ikke være tom"))
+            body("frontendFeilmelding", Is("Du må skrive en begrunnelse for endring av enhet"))
+        }
+    }
+
+    @Test
+    fun `endreBehandlendeEnhet - skal returnere behandling med endret enhet`() {
+        val token = lokalTestToken(behandlerRolle = BehandlerRolle.BESLUTTER)
+        every { integrasjonClient.sjekkTilgangTilPersoner(any()) } returns Tilgang(true, "test")
+        every { integrasjonClient.hentNavKontorEnhet("50") } returns NavKontorEnhet(50, "nyNavn", "nyEnhetNr", "nyStatus")
+
+        Given {
+            header("Authorization", "Bearer $token")
+            contentType(ContentType.JSON)
+            body(objectMapper.writeValueAsString(EndreBehandlendeEnhetDto("50", "nybegrunnelse")))
+        } When {
+            put("$behandlingControllerUrl/${behandling.id}/enhet")
+        } Then {
+            statusCode(HttpStatus.OK.value())
+            body("data.arbeidsfordelingPåBehandling.behandlendeEnhetId", Is("50"))
+            body("data.arbeidsfordelingPåBehandling.behandlendeEnhetNavn", Is("nyNavn"))
+            body("data.arbeidsfordelingPåBehandling.manueltOverstyrt", Is(true))
+        }
+    }
+
+    private fun lagreFagsak(): Fagsak {
+        val aktør = aktørRepository.saveAndFlush(randomAktør())
+
+        return fagsakRepository.saveAndFlush(lagFagsak(aktør))
+    }
+
+    private fun lagreBehandling(fagsak: Fagsak): Behandling {
+        val behandling =
+            behandlingRepository.saveAndFlush(lagBehandling(fagsak, opprettetÅrsak = BehandlingÅrsak.SØKNAD))
+
+        arbeidsfordelingPåBehandlingRepository.save(
+            ArbeidsfordelingPåBehandling(
+                behandlingId = behandling.id,
+                behandlendeEnhetId = "test",
+                behandlendeEnhetNavn = "test"
+            )
+        )
+
+        return behandling
+    }
```

Comment:
Hva tenker dere om å flytte disse inn til OppslagSpringRunnerTest? 
Ser at i nesten alle integrasjonstester så er vi avhengige av at det finnes fagsak + behandling + aktør.
Kan vi ikke bare lage en metode ala settOppInitiellData i OppslagSpringRunnerTest som setter opp dette for oss?

Reply:
Høres ut som en veldig god plan! Støttes! 😄